### ISSUE TITLE
pythonic_params does not cover the path params

### ISSUE THREAD

**Issue by kornicameister** [zalando/connexion#687]
### Description

`pythonic_params` allows to convert `query_params` (for instance) to more Pythonic form. 
For instance `testId`, in Python is `test_id`. However that is not reflected for path params.

### Expected behaviour

path params are subject to `pythonic_params=True`. 

### Actual behaviour

path params are not converted to snake case for python functions.

### Steps to reproduce

1. Take https://github.com/zalando/connexion/tree/master/examples/helloworld
2. Modify swagger to have `{name}` -> `{someName`}
3. Modify corresponding function to have `{name}` -> `{some_name}` in its declaration
3. Set `pythonicParams=True` when creating an app
4. Request fails because it can't match the params to function

### Additional info:

Output of the commands:

- `python --version` -> `3.6.5`
- `pip show connexion | grep "^Version\:"` -> `1.5.3`

**Comment by hjacobs** [comment]
Makes sense, I think the current behaviour is due to relying on Flask's route handling.

**Comment by dtkav** [comment]
In the case of query parameters the `key` is passed in by the user (`?myParam=5`), but with path parameters this isn't the case (`/5/`). What's the benefit of having the path parameter be camelCase? Consistency in documentation?

**Comment by hjacobs** [comment]
@dtkav I guess you are right, but one reason could be that the Swagger file is not completely under control of the implementor, e.g. that the Swagger/OpenAPI YAML needs to follow certain standards/consistency which would not break by using Connexion as the implementation server software.

**Comment by kornicameister** [comment]
Yeah, actually I wanted to report that most likely I messed up on my end. You see, I am using `telepresence` to deploy the dev service to `k8s` cluster and it sometimes chokes (i.e. does not immediately refresh the state on cluster). It seems like that was what happened to me because today I was finishing the PR in my project and as you @dtkav said I had the path parameters written in camelCase for the consistency. But then again I had to adjust multiple pieces of the code due to invalid keyword argument errors. Just to give an example.

I had path argument `id`, so I renamed that to `someId`. Until I changed to handler function declaration to say `some_id` code was failing.

So I think the conclusion we can draw from here is that at least for the FlaskApp, it seems as if path params are converted to snake_case dynamically. I feel quite safe to say this after today's refactor of my code. Should we close this down then? If so, sorry for the hassle, obviously speaking that was my fault :(.
